### DISCUSSION TITLE
Adapter Yaml settings

### DISCUSSION THREAD

**Discussion by matrix4767** [Mikubill/sd-webui-controlnet#179] [Q&A] [answered]
What should the config for these files be to work with models? They give an error otherwise and the webui proceeds to generate as if it's disabled.

**Comment by Mikubill** [comment, marked as answer]
sketch -> /path/sketch_adapter_v14.yaml
keypose or segmentation -> /path/image_adapter_v14.yaml

**Reply by matrix4767** [reply]
I'm not quire sure what you mean by that. But here's what happens when generating with keypose:
Traceback (most recent call last):
  File "G:\stable-webui\modules\scripts.py", line 386, in process
    script.process(p, *script_args)
  File "G:\stable-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 403, in process
    network = network_module(
  File "G:\stable-webui\extensions\sd-webui-controlnet\scripts\adapter.py", line 35, in __init__
    self.control_model.load_state_dict(state_dict)
  File "G:\stable-webui\venv\lib\site-packages\torch\nn\modules\module.py", line 1671, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for Adapter:
        size mismatch for conv_in.weight: copying a param with shape torch.Size([320, 192, 3, 3]) from checkpoint, the shape in current model is torch.Size([320, 64, 3, 3]).

**Reply by Mikubill** [reply]
It means you need to modify the `Config file for Adapter models` in the settings, change the config file name to image_adapter_v14.yaml, and it will work as usual.

**Reply by matrix4767** [reply]
Ah, I see. So the adapter yaml in the settings needs to be changed to whatever model is going to be used because they're separate files. Thanks.

**Reply by specblades** [reply]
Where i can find preprocessor for Adapter Pose model?
Piginet produces only sketches.
<img width="298" alt="image" src="https://user-images.githubusercontent.com/113993746/219871181-825f1481-a36b-4933-808f-77cb584845cc.png">

**Reply by gostyshev-e** [reply]
Is there additional .yaml configs will be available for updated sd15v1 and ad15v2 adapters? Also some thoughts on coadapters compatibility please? 
![image](https://user-images.githubusercontent.com/109570081/228335306-5c4929a3-3afe-492e-83d6-8fa8130311be.png)

**Reply by mlwei1** [reply]
> It means you need to modify the `Config file for Adapter models` in the settings, change the config file name to image_adapter_v14.yaml, and it will work as usual.

how about there is no image_adapter_v14.yaml  and sketch_adapter_v14.yaml,  the two is missing, does some site provide  this yaml?

**Reply by ljleb** [reply]
They have been moved:
- `sketch_adapter_v14.yaml` -> `t2iadapter_sketch_sd14v1.yaml`
- `image_adapter_v14.yaml` -> `t2iadapter_zoedepth_sd15v1.yaml`

I don't think we intend to have everybody manually update the config in the settings each time the model is changed, I think we need to update the code to make it work automatically if it is not already implemented in latest.

**Reply by mlwei1** [reply]
> They have been moved:
> 
> * `sketch_adapter_v14.yaml` -> `t2iadapter_sketch_sd14v1.yaml`
> * `image_adapter_v14.yaml` -> `t2iadapter_zoedepth_sd15v1.yaml`
> 
> I don't think we intend to have everybody manually update the config in the settings each time the model is changed, I think we need to update the code to make it work automatically if it is not already implemented in latest.

thx for answer, now i set t2iadapter_sketch_sd15v2.yaml in config, and it work with some 15v2 models

**Reply by Macroeye** [reply]
I lost two toolbars  Config file for Control Net models and  Config file for Adapter models after installing the latest controlnet

stable-diffusion-webui[Version: v1.7.0]
sd-webui-controlnet[v1.1.424]
OS[win]
Browser[Microsoft Edge]

**Comment by dpapperini** [comment]
At the following link [https://github.com/Mikubill/sd-webui-controlnet/tree/main/models](https://github.com/Mikubill/sd-webui-controlnet/tree/main/models), you can find the **.yaml files**.

At the following link [https://huggingface.co/TencentARC/T2I-Adapter/tree/main/models](https://huggingface.co/TencentARC/T2I-Adapter/tree/main/models), you can find the **.pth files**.

After downloading all the .yaml and .pth files and placing them into the model folder /stable-diffusion-webui/extensions/sd-webui-controlnet/models, I performed a machine reboot and relaunched the webui.

When I navigate to the ControlNet tab and click on the Models section, I can see all the .pth files listed. However, when I click on the Preprocessor dropdown menu, not all the .yaml files are visible.

Here is the list of files present in the folder /stable-diffusion-webui/extensions/sd-webui-controlnet/models:
```
cldm_v15.yaml
cldm_v21.yaml
coadapter-canny-sd15v1.pth
coadapter-color-sd15v1.pth
coadapter-depth-sd15v1.pth
coadapter-fuser-sd15v1.pth
coadapter-sketch-sd15v1.pth
coadapter-style-sd15v1.pth
control-net-v15-pool.yaml
control-net-v15.yaml
control_sd15_canny.pth
control_sd15_canny.yaml
control_sd15_depth.pth
control_sd15_depth.yaml
control_sd15_hed.pth
control_sd15_hed.yaml
control_sd15_mlsd.pth
control_sd15_mlsd.yaml
control_sd15_normal.pth
control_sd15_normal.yaml
control_sd15_openpose.pth
control_sd15_openpose.yaml
control_sd15_scribble.pth
control_sd15_scribble.yaml
control_sd15_seg.pth
control_sd15_seg.yaml
control_v11e_sd15_ip2p.pth
control_v11e_sd15_ip2p.yaml
control_v11e_sd15_shuffle.pth
control_v11e_sd15_shuffle.yaml
control_v11f1e_sd15_tile.pth
control_v11f1e_sd15_tile.yaml
control_v11f1p_sd15_depth.pth
control_v11f1p_sd15_depth.yaml
control_v11p_sd15_canny.pth
control_v11p_sd15_canny.yaml
control_v11p_sd15_inpaint.pth
control_v11p_sd15_inpaint.yaml
control_v11p_sd15_lineart.pth
control_v11p_sd15_lineart.yaml
control_v11p_sd15_mlsd.pth
control_v11p_sd15_mlsd.yaml
control_v11p_sd15_normalbae.pth
control_v11p_sd15_normalbae.yaml
control_v11p_sd15_openpose.pth
control_v11p_sd15_openpose.yaml
control_v11p_sd15s2_lineart_anime.pth
control_v11p_sd15s2_lineart_anime.yaml
control_v11p_sd15_scribble.pth
control_v11p_sd15_scribble.yaml
control_v11p_sd15_seg.pth
control_v11p_sd15_seg.yaml
control_v11p_sd15_softedge.pth
control_v11p_sd15_softedge.yaml
control_v11p_sd21_qrcode.safetensors
control_v11p_sd21_qrcode.yaml
control_v1p_sd15_brightness.safetensors
control_v1p_sd15_illumination.safetensors
control_v1p_sd15_qrcode.safetensors
control_v1p_sd15_qrcode.yaml
diff_control_sd15_temporalnet_fp16.safetensors
image_adapter_v14.yaml
instruct-pix2pix.yaml
sketch_adapter_v14.yaml
t2iadapter_canny_sd14v1.pth
t2iadapter_canny_sd14v1.yaml
t2iadapter_canny_sd15v2.pth
t2iadapter_canny_sd15v2.yaml
t2iadapter_color_sd14v1.pth
t2iadapter_color_sd14v1.yaml
t2iadapter_depth_sd14v1.pth
t2iadapter_depth_sd14v1.yaml
t2iadapter_depth_sd15v2.pth
t2iadapter_depth_sd15v2.yaml
t2iadapter_keypose_sd14v1.pth
t2iadapter_keypose_sd14v1.yaml
t2iadapter_openpose_sd14v1.pth
t2iadapter_openpose_sd14v1.yaml
t2iadapter_seg_sd14v1.pth
t2iadapter_seg_sd14v1.yaml
t2iadapter_sketch_sd14v1.pth
t2iadapter_sketch_sd14v1.yaml
t2iadapter_sketch_sd15v2.pth
t2iadapter_sketch_sd15v2.yaml
t2iadapter_style_sd14v1.pth
t2iadapter_style_sd14v1.yaml
t2iadapter_zoedepth_sd15v1.pth
t2iadapter_zoedepth_sd15v1.yaml
```
I have attached screenshots of the two dropdown menus in ControlNet for reference.
![Screenshot 2023-07-20 alle 14 58 01](https://github.com/TencentARC/T2I-Adapter/assets/77624959/7caa2c89-1071-41c9-9234-c4a8966b343e)
![Screenshot 2023-07-20 alle 14 57 51](https://github.com/TencentARC/T2I-Adapter/assets/77624959/25ebe00e-a319-46df-b765-a50bd7e9b093)

Do you have any idea why this might be happening and how to resolve the issue?
